File: NirvanaSMS.Api.CSharp/Core/Base/HeaderBase.cs
using NirvanaSMS.Api.Core.Interface;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class HeaderBase : IHeader
    {
        public abstract ProcessResult<string[]> Query(ApiUser apiUser);
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/HeaderBase.java
package org.nirvanasms.api.core.base;

import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.interfaces.IHeader;
import org.nirvanasms.api.core.utility.ProcessResult;

public abstract  class HeaderBase implements IHeader {

	@Override
	public abstract ProcessResult<String[]> query(ApiUser apiUser);
}
